frontend/dockerfile/dockerfile_addgit_test.go

```diff
@@ -350,6 +352,239 @@ COPY --from=src /repo/unique.txt /
 	require.Equal(t, string(unique1), string(unique2), "cache should be matched and unique file content should be the same")
 }
 
+func testGitQueryString(t *testing.T, sb integration.Sandbox) {
+	integration.SkipOnPlatform(t, "windows")
+	f := getFrontend(t, sb)
+
+	gitDir, err := os.MkdirTemp("", "buildkit")
+	require.NoError(t, err)
+	defer os.RemoveAll(gitDir)
+	err = runShell(gitDir, []string{
+		"git init",
+		"git config --local user.email test",
+		"git config --local user.name test",
+		"echo base >foo",
+	}...)
+	require.NoError(t, err)
+
+	err = os.WriteFile(filepath.Join(gitDir, "Dockerfile"), []byte(`
+FROM scratch
+COPY foo out
+`), 0600)
+	require.NoError(t, err)
+
+	err = runShell(gitDir, []string{
+		"git add Dockerfile foo",
+		"git commit -m initial",
+		"git tag v0.0.1",
+		"git branch base",
+		"echo feature >foo",
+		"mkdir sub",
+		"echo subfeature >sub/foo",
+		"cp Dockerfile sub/",
+		"git add foo sub",
+		"git commit -m feature",
+		"git branch feature",
+		"git checkout -B master base",
+		"echo v0.0.2 >foo",
+		"git add foo",
+		"git commit -m v0.0.2",
+		"git tag v0.0.2",
+		"echo latest >foo",
+		"git add foo",
+		"git commit -m latest",
+		"git tag latest",
+		"git update-server-info",
+	}...)
+	require.NoError(t, err)
+
+	// get commit SHA for v0.0.2
+	cmd := exec.Command("git", "rev-parse", "v0.0.2")
+	cmd.Dir = gitDir
+	dt, err := cmd.CombinedOutput()
+	require.NoError(t, err)
+	commitHashV2 := strings.TrimSpace(string(dt))
+	require.Len(t, commitHashV2, 40)
+
+	// get commit SHA for latest
+	cmd = exec.Command("git", "rev-parse", "latest")
+	cmd.Dir = gitDir
+	dt, err = cmd.CombinedOutput()
+	require.NoError(t, err)
+	commitHashLatest := strings.TrimSpace(string(dt))
+	require.Len(t, commitHashLatest, 40)
+	require.NotEqual(t, commitHashV2, commitHashLatest)
+
+	server := httptest.NewServer(http.FileServer(http.Dir(filepath.Clean(gitDir))))
+	defer server.Close()
+	serverURL := server.URL
+
+	c, err := client.New(sb.Context(), sb.Address())
+	require.NoError(t, err)
+	defer c.Close()
+
+	type tcase struct {
+		name      string
+		url       string
+		expectOut string
+		expectErr string
+	}
+
+	tcases := []tcase{
+		{
+			// if this commit is already cached then this will work and ignore tag atm because tag name has no influence to the output
```

Comment:
@AkihiroSuda Bit unexpected, but I don't think there is another way. This is not any new behavior from querystring but same for `--checksum`.